diff --git a/src/virtualbox6/src/VBox/Devices/Audio/DevHda.cpp b/src/virtualbox6/src/VBox/Devices/Audio/DevHda.cpp
index 4e87984..0377828 100644
--- a/src/virtualbox6/src/VBox/Devices/Audio/DevHda.cpp
+++ b/src/virtualbox6/src/VBox/Devices/Audio/DevHda.cpp
@@ -4564,8 +4564,8 @@ static int hdaR3AttachInternal(PPDMDEVINS pDevIns, PHDASTATE pThis, PHDASTATER3
                 Cfg.enmPath                       = PDMAUDIOPATH_OUT_FRONT;
                 Cfg.Device.cMsSchedulingHint      = 10;
                 Cfg.Backend.cFramesPreBuffering   = UINT32_MAX;
-                PDMAudioPropsInit(&Cfg.Props, 2, true /*fSigned*/, 2, 44100);
-                RTStrPrintf(Cfg.szName, sizeof(Cfg.szName), "output 44.1kHz 2ch S16 (HDA config hint)");
+                PDMAudioPropsInit(&Cfg.Props, 2, true /*fSigned*/, 2, 48000);
+                RTStrPrintf(Cfg.szName, sizeof(Cfg.szName), "output 48kHz 2ch S16 (HDA config hint)");
 
                 pDrv->pConnector->pfnStreamConfigHint(pDrv->pConnector, &Cfg); /* (may trash CfgReq) */
             }
diff --git a/src/virtualbox6/src/VBox/Devices/Audio/DevHdaCodec.cpp b/src/virtualbox6/src/VBox/Devices/Audio/DevHdaCodec.cpp
index 0c86572..2a3f3fe 100644
--- a/src/virtualbox6/src/VBox/Devices/Audio/DevHdaCodec.cpp
+++ b/src/virtualbox6/src/VBox/Devices/Audio/DevHdaCodec.cpp
@@ -227,7 +227,7 @@ static void stac9220NodeReset(PHDACODECR3 pThis, uint8_t uNID, PCODECNODE pNode,
         {
             pNode->afg.node.au32F00_param[0x08] = CODEC_MAKE_F00_08(1, 0xd, 0xd);
             /* We set the AFG's PCM capabitilies fixed to 16kHz, 22.5kHz + 44.1kHz, 16-bit signed. */
-            pNode->afg.node.au32F00_param[0x0A] = CODEC_F00_0A_44_1KHZ          /* 44.1 kHz */
+            pNode->afg.node.au32F00_param[0x0A] = CODEC_F00_0A_48KHZ
                                                 | CODEC_F00_0A_44_1KHZ_1_2X     /* Messed up way of saying 22.05 kHz */
                                                 | CODEC_F00_0A_48KHZ_1_3X       /* Messed up way of saying 16 kHz. */
                                                 | CODEC_F00_0A_16_BIT;          /* 16-bit signed */
@@ -848,7 +848,7 @@ static int stac9220Construct(PHDACODECR3 pThis, HDACODECCFG *pCfg)
     /* Common AFG node initializers. */
     pThis->aNodes[STAC9220_NID_AFG].afg.node.au32F00_param[0x4] = CODEC_MAKE_F00_04(0x2, STAC9221_NUM_NODES - 2);
     pThis->aNodes[STAC9220_NID_AFG].afg.node.au32F00_param[0x5] = CODEC_MAKE_F00_05(1, CODEC_F00_05_AFG);
-    pThis->aNodes[STAC9220_NID_AFG].afg.node.au32F00_param[0xA] = CODEC_F00_0A_44_1KHZ | CODEC_F00_0A_16_BIT;
+    pThis->aNodes[STAC9220_NID_AFG].afg.node.au32F00_param[0xA] = CODEC_F00_0A_48KHZ | CODEC_F00_0A_16_BIT;
     pThis->aNodes[STAC9220_NID_AFG].afg.u32F20_param = CODEC_MAKE_F20(pCfg->idVendor, pCfg->bBSKU, pCfg->idAssembly);
 
     return VINF_SUCCESS;
