diff --git src/virtualbox7/src/VBox/VMM/VMMR3/target-x86/CPUMR3CpuId-x86.cpp src/virtualbox7/src/VBox/VMM/VMMR3/target-x86/CPUMR3CpuId-x86.cpp
index eafe6612..2c3c4e1b 100644
--- src/virtualbox7/src/VBox/VMM/VMMR3/target-x86/CPUMR3CpuId-x86.cpp
+++ src/virtualbox7/src/VBox/VMM/VMMR3/target-x86/CPUMR3CpuId-x86.cpp
@@ -3086,7 +3086,7 @@ static int cpumR3CpuIdReadConfig(PVM pVM, PCPUMCPUIDCONFIG pConfig, PCFGMNODE pC
      * the default is to only do this for VMs with nested paging and AMD-V or
      * unrestricted guest mode.
      */
-    rc = cpumR3CpuIdReadIsaExtCfg(pVM, pIsaExts, "RDRAND", &pConfig->enmRdRand, fNestedPagingAndFullGuestExec);
+    rc = cpumR3CpuIdReadIsaExtCfg(pVM, pIsaExts, "RDRAND", &pConfig->enmRdRand, fEnforceHWusage);
     AssertLogRelRCReturn(rc, rc);
 
     /** @cfgm{/CPUM/IsaExts/RDSEED, isaextcfg, depends}
@@ -3094,7 +3094,7 @@ static int cpumR3CpuIdReadConfig(PVM pVM, PCPUMCPUIDCONFIG pConfig, PCFGMNODE pC
      * the default is to only do this for VMs with nested paging and AMD-V or
      * unrestricted guest mode.
      */
-    rc = cpumR3CpuIdReadIsaExtCfg(pVM, pIsaExts, "RDSEED", &pConfig->enmRdSeed, fNestedPagingAndFullGuestExec);
+    rc = cpumR3CpuIdReadIsaExtCfg(pVM, pIsaExts, "RDSEED", &pConfig->enmRdSeed, fEnforceHWusage);
     AssertLogRelRCReturn(rc, rc);
 
     /** @cfgm{/CPUM/IsaExts/ADX, isaextcfg, depends}
