Genode's libc does not export _setjmp/_longjmp

diff --git src/virtualbox7/include/iprt/setjmp-without-sigmask.h src/virtualbox7/include/iprt/setjmp-without-sigmask.h
index c78603a4..e1984053 100644
--- src/virtualbox7/include/iprt/setjmp-without-sigmask.h
+++ src/virtualbox7/include/iprt/setjmp-without-sigmask.h
@@ -53,7 +53,8 @@
      || defined(RT_OS_DRAGONFLY) \
      || defined(RT_OS_FREEBSD) \
      || defined(RT_OS_NETBSD) \
-     || defined(RT_OS_OPENBSD) )
+     || defined(RT_OS_OPENBSD)) \
+ && !defined(__GENODE__)
 # undef setjmp /* /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/setjmp.h defines it on macOS. */
 # define setjmp  _setjmp
 # define longjmp _longjmp
