Limit EMT stack size to prevent warning.

--- src/virtualbox7/src/VBox/VMM/VMMR3/VM.cpp
+++ src/virtualbox7/src/VBox/VMM/VMMR3/VM.cpp
@@ -512,7 +512,7 @@ static int vmR3CreateUVM(uint32_t cCpus, PCVMM2USERMETHODS pVmm2UserMethods, PUV
                                 for (i = 0; i < cCpus; i++)
                                 {
                                     rc = RTThreadCreateF(&pUVM->aCpus[i].vm.s.ThreadEMT, vmR3EmulationThread, &pUVM->aCpus[i],
-                                                         _1M, RTTHREADTYPE_EMULATION,
+                                                         _1M - _16K, RTTHREADTYPE_EMULATION,
                                                          RTTHREADFLAGS_WAITABLE | RTTHREADFLAGS_COM_MTA | RTTHREADFLAGS_NO_SIGNALS,
                                                          cCpus > 1 ? "EMT-%u" : "EMT", i);
                                     if (RT_FAILURE(rc))
