diff --git src/virtualbox7/src/VBox/Storage/VMDK.cpp src/virtualbox7/src/VBox/Storage/VMDK.cpp
index d62e3757..9c6efbdd 100644
--- src/virtualbox7/src/VBox/Storage/VMDK.cpp
+++ src/virtualbox7/src/VBox/Storage/VMDK.cpp
@@ -48,6 +48,17 @@
 #include <iprt/string.h>
 #include <iprt/zip.h>
 #include <iprt/asm.h>
+
+#ifdef __GENODE__
+/*
+ * Genode does not support vmdkCreateImage/vmdkMakeRawDescriptor and the copying
+ * of partition tables from a real disk into a vmdk and does not provide the
+ * libgeom library of FreeBSD. So we have to undefine it here because it will be
+ * defined through iprt/cdefs.h.
+ */
+#undef RT_OS_FREEBSD
+#endif
+
 #ifdef RT_OS_WINDOWS
 # include <iprt/utf16.h>
 # include <iprt/uni.h>
